Classes/Eel/PropTypesHelper.php

```diff
@@ -1,35 +1,186 @@
+    {
+        return (new PropTypeValidator())->regex($regularExpression);
+    }
+
+    /**
+     * Validate the value equals one of the given options.
+     *
+     * @example PropTypes.oneOf([123, "foo", "bar"])
+     *
+     * @param mixed[] $values
+     */
+    public function oneOf(array $values): PropTypeValidator
+    {
+        return (new PropTypeValidator())->oneOf($values);
+    }
+
+    /**
+     * Validate an array was given and all validate with the given validator, accepts null.
+     *
+     * @example PropTypes.arrayOf(PropTypes.string)
      */
+    public function arrayOf(ValidatorInterface $itemValidator): PropTypeValidator
+    {
+        return (new PropTypeValidator())->arrayOf($itemValidator);
+    }
+
+    /**
+     * Validate the value validates at least with one of the given validators, accepts null.
+     *
+     * @example PropTypes.anyOf( PropTypes.string, PropTypes.integer )
+     */
+    public function anyOf(ValidatorInterface ...$validators): PropTypeValidator
+    {
+        return (new PropTypeValidator())->anyOf(...$validators);
+    }
+
+    /**
+     * Validate the keys of the given array validate with the assigned Validator, accepts null and ignores all other keys.
+     *
+     * The key validators have to define wether a single key is required.
+     *
+     * alias for @see shape
+     *
+     * @example PropTypes.dataStructure({'foo': PropTypes.integer, 'bar': PropTypes.string})
+     *
+     * @param array $shape
+     */
+    public function dataStructure($shape): PropTypeValidator
+    {
+        return (new PropTypeValidator())->dataStructure($shape);
+    }
+
+    /**
+     * Validate the keys of the given array validate with the assigned Validator, accepts null and ignores all other keys.
+     * The key validators have to define wether a single key is required.
+     *
+     * alias for @see dataStructure
+     *
+     * @example PropTypes.shape({'foo': PropTypes.integer, 'bar': PropTypes.string})
+     *
+     * @param array $shape
+     */
+    public function shape($shape): PropTypeValidator
+    {
+        return (new PropTypeValidator())->shape($shape);
+    }
+
+    /**
+     * Validate that a given value is an existing file.
+     *
+     * @example PropTypes.fileExists
+     */
+    public function getFileExists(): PropTypeValidator
+    {
+        return (new PropTypeValidator())->getFileExists();
+    }
+
+    /**
+     * Validate the value with the given type, if the value is a Node the NodeType is checked instead of the php-class, accepts null.
+     *
+     * @example PropTypes.instanceOf('Neos.Neos:Document')
+     *
+     * @param string $type
+     */
+    public function instanceOf($type): PropTypeValidator
+    {
+        return (new PropTypeValidator())->instanceOf($type);
+    }
+
     public function allowsCallOfMethod($methodName)
     {
-        return in_array($methodName, PropTypeValidator::VALID_EEL_METHODS);
+        return true;
```

Comment:
fyi - currently and previously it was allowed to call `${PropTypes.getString()}` instead of `${PropTypes.string}` one could restrict that by no allowing `getString` to be called here.

The property access notation would still work, as `ObjectAccess::getProperty` doesnt check for `allowsCallOfMethod`